tests/test_username_api.py

```diff
@@ -40,306 +41,62 @@ def get_expected_response(website, user, status):
-
-      user = generate_random_username(15)
-      expected = get_expected_response('instagram', user, 404)
-      actual = get_response(self.app, 'instagram', user)
-
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['twitter']['available_usernames'])
-  def test_available_twitter_username(self, user):
-    expected = get_expected_response('twitter', user, 404)
-    actual = get_response(self.app, 'twitter', user)
-
-    message = None
-    if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
-
-      user = generate_random_username(15)
-      expected = get_expected_response('twitter', user, 404)
-      actual = get_response(self.app, 'twitter', user)
-
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['facebook']['available_usernames'])
-  def test_available_facebook_username(self, user):
-    expected = get_expected_response('facebook', user, 404)
-    actual = get_response(self.app, 'facebook', user)
-
-    # this is special to facebook checking,
-    # and 'profile' field is only useful for
-    # frontend purposes. Skipping here.
-    del actual['profile']
-
-    message = None
-    if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
-
-      user = generate_random_username(25)
-      expected = get_expected_response('facebook', user, 404)
-      actual = get_response(self.app, 'facebook', user)
+  @parameterized.expand(load_test_cases('available_username'), testcase_func_name=custom_name_func)
+  def test_available_name(self, website, user):
+    expected = get_expected_response(website, user, 404)
+    actual = get_response(self.app, website, user)
 
+    if website == 'facebook':
       del actual['profile']
 
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['openhub']['available_usernames'])
-  def test_available_openhub_username(self, user):
-    expected = get_expected_response('openhub', user, 404)
-    actual = get_response(self.app, 'openhub', user)
-
     message = None
     if expected != actual:
       message = 'The provided available username ({}) returned 200'.format(user)
```

Comment:
Is there any point in having these messages now? We can already see the used username with `parameterized`, and know that it's failing...

Reply:
Late reply.

Firstly, very interesting review comment.

Personally, I would not have recommended this change because it means the test result message isnt self-contained, and doesnt really make sense without the verbose format, or another plugin which uses the test function name in the output.

I think we could also uses a test function docstring generator to make this change compliant with [`pytest-spec`](https://pypi.python.org/pypi/pytest-spec) , but ensuring compatibility with that would be optional for this task.

I am not going to reject the task based on this, as the goal has been reached and this change was requested, reasonable and done correctly.  It definitely isnt 'wrong'.  I suspect this is a 'matter of taste' and I will defer to @manu-chroma on this one.

Reply:
ok

Reply:
As discussed in Zulip chat before, ``parameterized`` sanitises the usernames before creating calling ``custom_func_name``
For eg: When testing for available username: ``manu.chroma`` for any website: 
```
tests/test_username_api.py::TestUsernameApi::test_available_username_twitter_manu_chroma test_username_api.py       
122 WARNING  The provided available username (manu.chroma) returned 200 but mLuogoEZ0qEwl68 is still available
```
There is a chance of not seeing the actual username in the function name. I think it is for the better that we keep this logging logic for now